tests/console/commands/test_install.py

```diff
@@ -4,32 +4,122 @@
+readme = "README.rst"
+
+[tool.poetry.dependencies]
+python = "~2.7 || ^3.4"
+
+[tool.poetry.group.foo.dependencies]
+foo = "^1.0"
+
+[tool.poetry.group.bar.dependencies]
+bar = "^1.1"
+
+[tool.poetry.group.baz.dependencies]
+baz = "^1.2"
+
+[tool.poetry.group.bim.dependencies]
+bim = "^1.3"
+
+[tool.poetry.group.bam]
+optional = true
+
+[tool.poetry.group.bam.dependencies]
+bam = "^1.4"
+"""
 
 
 @pytest.fixture
-def tester(command_tester_factory: CommandTesterFactory) -> CommandTester:
+def poetry(project_factory: ProjectFactory) -> Poetry:
+    return project_factory(name="export", pyproject_content=PYPROJECT_CONTENT)
+
+
+@pytest.fixture
+def tester(
+    command_tester_factory: CommandTesterFactory, poetry: Poetry
+) -> CommandTester:
     return command_tester_factory("install")
 
 
+@pytest.mark.parametrize(
+    ("options", "groups", "with_root"),
+    [
+        case + (with_root,)
+        for case in [
+            ("", {"default", "foo", "bar", "baz", "bim"}),
+            ("--only default", {"default"}),
+            ("--only foo", {"foo"}),
+            ("--only foo,bar", {"foo", "bar"}),
+            ("--only bam", {"bam"}),
+            ("--with bam", {"default", "foo", "bar", "baz", "bim", "bam"}),
+            ("--without foo,bar", {"default", "baz", "bim"}),
+            ("--without default", {"foo", "bar", "baz", "bim"}),
+            ("--with foo,bar --without baz --without bim --only bam", {"bam"}),
+            # net result zero options
+            ("--with foo", {"default", "foo", "bar", "baz", "bim"}),
+            ("--without bam", {"default", "foo", "bar", "baz", "bim"}),
+            ("--with bam --without bam", {"default", "foo", "bar", "baz", "bim"}),
+            ("--with foo --without foo", {"default", "bar", "baz", "bim"}),
+            # deprecated options
+            ("--default", {"default"}),
+            ("--no-dev", {"default"}),
+            ("--dev-only", {"foo", "bar", "baz", "bim"}),
+        ]
+        for with_root in {True, False}
+    ],
+)
 def test_group_options_are_passed_to_the_installer(
-    tester: CommandTester, mocker: MockerFixture
+    options: str,
+    groups: set[str],
+    with_root: bool,
+    tester: CommandTester,
+    mocker: MockerFixture,
 ):
     """
     Group options are passed properly to the installer.
     """
-    mocker.patch.object(tester.command.installer, "run", return_value=1)
+    mocker.patch.object(tester.command.installer, "run", return_value=0)
+    editable_builder_mock = mocker.patch(
+        "poetry.masonry.builders.EditableBuilder", side_effect=ModuleOrPackageNotFound()
+    )
+
+    if not with_root:
+        options = f"--no-root {options}"
+
+    tester.execute(options)
+
+    package_groups = set(tester.command.poetry.package._dependency_groups.keys())
+    installer_groups = set(tester.command.installer._groups)
 
-    tester.execute("--with foo,bar --without baz --without bim --only bam")
+    assert installer_groups <= package_groups
```

Comment:
Shouldn't it be `len(installer_groups) <= len(package_groups)`?

Reply:
Why? If you do len you won't be verifying the values, just the count. Current assertion is that the installed groups are a subset of the project groups.

Reply:
I simply wasn't aware of this syntax. Maybe use `issubset()` instead?

Reply:
I don't feel that is required per say, let me know if you feel strongly about it.